### ISSUE TITLE
Microk8s reset stuck after "Warning Ingress is deprecated" message

### ISSUE THREAD

**Issue by artues** [ubuntu/microk8s#1609]
<h1>Version</h1>
v1.19.2-34+1b3fa60b402c1c

installed using snap --classic

<h1>Issue</h1>
I cant reset my microk8s node, it get stuck after some warnings with the same message and I need to ctrl + c

[inspection-report-20200929_194702.tar.gz](https://github.com/ubuntu/microk8s/files/5301461/inspection-report-20200929_194702.tar.gz)

```
arturo@dred:~$ microk8s reset
Disabling all addons.
Disabling addon : ambassador
Disabling addon : cilium
Disabling addon : dashboard
Disabling addon : dns
Disabling addon : fluentd
Disabling addon : gpu
Disabling addon : helm
Disabling addon : helm3
Disabling addon : host-access
Disabling addon : ingress
Disabling addon : istio
Disabling addon : jaeger
Disabling addon : juju
Disabling addon : knative
Disabling addon : kubeflow
Disabling addon : linkerd
Disabling addon : metallb
Disabling addon : metrics-server
Disabling addon : multus
Disabling addon : prometheus
Disabling addon : rbac
Disabling addon : registry
Disabling addon : registry-help
Disabling addon : storage
All addons disabled.
Calling clean_cluster
Cleaning resources in namespace kube-system
No resources found
No resources found
No resources found
configmap "extension-apiserver-authentication" deleted
endpoints "kube-controller-manager" deleted
endpoints "kube-scheduler" deleted
lease.coordination.k8s.io "kube-controller-manager" deleted
lease.coordination.k8s.io "kube-scheduler" deleted
Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
Cleaning resources in namespace kube-public
No resources found
No resources found
No resources found
Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
No resources found
Cleaning resources in namespace kube-node-lease
No resources found
No resources found
No resources found
lease.coordination.k8s.io "dred" deleted
Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
Cleaning resources in namespace default
No resources found
No resources found
No resources found
endpoints "kubernetes" deleted
persistentvolumeclaim "my-release-moodle" deleted
persistentvolumeclaim "data-my-release-mariadb-0" deleted
service "kubernetes" deleted
endpointslice.discovery.k8s.io "kubernetes" deleted
Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress

```

**Comment by gliptak** [comment]
current PR build also fails with similar error https://github.com/ubuntu/microk8s/pull/1616/checks?check_run_id=1193543738

**Comment by balchua** [comment]
@gliptak thanks for reporting this issue.  It looks like it is stuck in deleting the persistent volume claims.

```shell
NAME                        STATUS        VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   AGE
my-release-moodle           Terminating                                                     142m
data-my-release-mariadb-0   Terminating                                                     142m
```
I usually just do a `sudo snap remove microk8s --purge` if i want to nuke my cluster. Then followed by `sudo snap install microk8s --classic --channel [your preferred channel]`
This one does the job cleanly.

**Comment by balchua** [comment]
@ktsakalozos  While trying to reproduce this last weekend, i found out that in certain cases `microk8s reset` doesn't restore back the cni.
Because `microk8s start` returns `1` so the script exists before reaching the point where we apply the `cni.yaml`.
For some strange reason it looks like it fails here.
https://github.com/ubuntu/microk8s/blob/8c23ed3800608aa10c7b6d6b01ac9c256f382464/microk8s-resources/actions/common/utils.sh#L429

Didn't get the chance to actually print it.  It doesn't happen all the time though.  My guess is it could be related to the "checkpoint in progress" issue.  But i have to try it again.

**Comment by dkupper76** [comment]
I just experienced this issue, and I tried several actions to resolve it such as: opening another terminal, and reboot, then running the microk8s reset again, and then I would try opening another terminal and running microk8s status and/or microk8s inspect and would get results back such as microk8s is not running or docker installed, etc. (uninstalling docker didn't make any difference), and I don't recall installing it in the first place, I just installed microk8s and enabled add ons and started creating pods, etc., after much searching and trying several different options. Finally I took the advice of @balchua , and ran the command sudo snap remove microk8s --purge, which seems to be the only way out of this. Thank you @balchua for that suggestion to run the command sudo snap remove microk8s --purge.
